azkaban-common/src/main/java/azkaban/imagemgmt/daos/ImageVersionDaoImpl.java

```diff
@@ -251,6 +258,54 @@ public int createImageVersion(final ImageVersion imageVersion) {
     return imageVersions;
   }
 
+  @Override
+  public List<ImageVersion> getLatestNonActiveVersionByImageTypes(final Set<String> imageTypes)
+      throws ImageMgmtException {
+    List<ImageVersion> imageVersions = new ArrayList<>();
+    try {
+      // Add outer select clause
+      final StringBuilder inClauseBuilder = new StringBuilder();
+      for (int i = 0; i < imageTypes.size(); i++) {
+        inClauseBuilder.append("?,");
```

Comment:
String.join works well if we know the strings to be be appended before hand and they are fixed. In this case "?" is appended for each image type to form the in clause. If we use string.join in the loop it will create a new string every time which is expensive. On the other hand StringBuilder will append to the existing one and thus avoiding creation of new string.